[AGENT-16456] Fix Azure KeyVault azure_session config example (#37847)
DataDog/documentation: content/en/agent/configuration/secrets-management.md

secret_backend_type: azure.keyvault
secret_backend_config:
  keyvaulturl: {keyVaultURL}
  azure_session:

>     azure_client_id: {clientID}  # User-assigned managed identity client ID; omit for system-assigned

Reviewer: ```suggestion
    azure_client_id: {clientID}  # User-assigned managed identity client ID; omit this field for system-assigned
```

Small clarification

| `vault_password` | Password for userpass authentication. Use with `vault_username`. |
| `vault_ldap_username` | Username for LDAP authentication. Use with `vault_ldap_password`. |
| `vault_ldap_password` | Password for LDAP authentication. Use with `vault_ldap_username`. |
| `vault_aws_role` | Vault role name for AWS IAM authentication. Required when `vault_auth_type: aws`. |
| `vault_aws_iam_server_id` | Value for the `X-Vault-AWS-IAM-Server-ID` header, used to prevent replay attacks. |
| `aws_region` | AWS region for IAM authentication requests. Defaults to `us-east-1`. |
| `vault_kubernetes_role` | Vault role name for Kubernetes authentication. Required when `vault_auth_type: kubernetes`. |
| `vault_kubernetes_jwt` | Kubernetes service account JWT token as a string. |
| `vault_kubernetes_jwt_path` | Path to the Kubernetes JWT token file. Defaults to `/var/run/secrets/kubernetes.io/serviceaccount/token`. |
| `vault_kubernetes_mount_path` | Vault mount path for the Kubernetes auth method. |
| `implicit_auth` | Set to `true` to skip authentication and use the token already set in the Vault client environment (for example, `VAULT_TOKEN`). |

> ##### Other `vault_session` options

Reviewer: This looks like a correct flag to me, but let me know what you think

The following top-level `secret_backend_config` fields also apply:

| Field | Description |
|---|---|
| `vault_address` | Vault server address (for example, `http://myvaultaddress.net`). Can also be set with the `VAULT_ADDR` environment variable. |
| `vault_token` | Static Vault token. Use when not relying on an auth method. |
| `vault_namespace` | Vault namespace for Vault Enterprise environments. |

##### TLS configuration (`vault_tls_config`)

To enable mutual TLS or a custom CA, add a `vault_tls_config` block:

```yaml
secret_backend_type: hashicorp.vault
secret_backend_config:
  vault_address: https://myvaultaddress.net
  vault_tls_config:
    ca_cert: /path/to/ca.pem
    client_cert: /path/to/client.pem
    client_key: /path/to/client-key.pem
    insecure: false
```

| Field | Description |
|---|---|
| `ca_cert` | Path to a PEM-encoded CA certificate file. |
| `ca_path` | Path to a directory of PEM-encoded CA certificate files. |
| `client_cert` | Path to a PEM-encoded client certificate file for mTLS. |
| `client_key` | Path to the private key file for the client certificate. |
| `tls_server` | Expected server name for TLS SNI verification. |

> | `insecure` | Set to `true` to disable TLS certificate verification. Not recommended for production. |

Reviewer: ```suggestion
| `insecure` | Set to `true` to disable TLS certificate verification. Do not use in production. |
```

Optional strengthening
